src/main.ts

```diff
@@ -19,7 +19,7 @@ async function run(): Promise<void> {
       [
         '-B',
         '-X',
-        ...(core.getInput('maven-options')),
+        ...(core.getInput('maven-options').split(' ').trim()),
```

Comment:
like you already pointed out, I don't like the split here. the spaces in params will come back to bite us somewhere.

Just check the API, indeed, `getInput` only supports strings: https://github.com/actions/toolkit/blob/main/packages/core/README.md#inputsoutputs . There is this old feature request for something similar: https://github.com/actions/toolkit/issues/184

Here they discuss a nice alternative, a multiline value. so you split on newlines:

```yaml
maven-options: |
   -Darg1="with spaces"
   -Darg2=2G
   -Xmx5g
```

would work then.